File: 2026.2 Estrutura de dados/Exercicios/0.5 - avaliacao.py
def cadastrarVetor(vetor, qtdElementos):
    for i in range(qtdElementos):
        while(True):
            try:
                vetor[i] = int(input(f"Digite o valor inteiro da posição V[{i}]:\n"))
                break
            except ValueError:
                print("valor digitado precisa ser um inteiro! Digite novamente\n")
            except Exception:
                print("erro. Digite novamente\n")
    
def mostrarVetor(vetor, elementos):
    for i in range(elementos):
        print(f"posição V[{i}]: valor {vetor[i]}" if i != elementos - 1 else f"posição V[{i}]: valor {vetor[i]}\n\n")

def AlterarVetor(vetor, elementos, procurado, novoValor):
    alteracoes = 0
    for i in range(elementos):
        if (vetor[i] == procurado):
            vetor[i] = novoValor
            alteracoes+=1
    return alteracoes
    

if __name__ == "__main__":
    vetor = [0] * 10
    print(f"Bem vindo ao sistema de gerenciamento de vetores!")
    print(f"Para sair do sistema use CTRL + C!!\n\n")
    try:
        cadastrarVetor(vetor, 10)
        mostrarVetor(vetor, 10)
        totalAlterado = AlterarVetor(vetor, 10, 9, -1)
        mostrarVetor(vetor, 10)
        if (totalAlterado > 0):
            print(f"Elementos alterados com sucesso, foram alteradas {totalAlterado} poisições!")
        elif (totalAlterado<=0):
            print("O valor não foi enccontrado. Nenhuma posição foi alterada!")
    except KeyboardInterrupt:
        print("Obrigado por utilizar o sistema!\n")
    

File: 2026.2 Estrutura de dados/Exercicios/Ray/0.5 - avaliacao.c
#include <stdio.h>

int lerInteiro(char *mensagem) {
    int valor;
    while (1) {
        printf("%s", mensagem);
        if (scanf("%d", &valor) == 1) {
            break;
        } else {
            printf("Valor inválido! Digite um número inteiro.\n\n");
            while (getchar() != '\n');
        }
    }
    return valor;
}   

void cadastrarVetor(int vetor[], int elementos) {
    for (int i = 0; i < elementos; i++) {
        char mensagem[100];
        sprintf(mensagem, "Digite o valor do elemento na posição V[%d]\n", i);
        vetor[i] = lerInteiro(mensagem);  
    }
}

void mostrarVetor(int vetor[], int elementos) {
    for (int i = 0; i < elementos; i++) {
        if (i != elementos - 1) {
            printf("Posição V[%d]: valor %d\n", i, vetor[i]);
        } else {
            printf("Posição V[%d]: valor %d\n\n", i, vetor[i]);
        }
    }
}

int alterarVetor(int vetor[], int elementos, int procurado, int novoValor) {
    int elementosAlterados = 0;
    for (int i = 0; i < elementos; i++) {
        if (vetor[i] == procurado) {
            vetor[i] = novoValor;
            elementosAlterados++;
        }
    }
    return elementosAlterados;
}

void exibirAcoes() {
    printf("Operações disponíveis para realizar com seu vetor ou CTRL + C para SAIR!!\n");
    printf("1 - Alterar Vetor\n");
    printf("2 - Mostrar Vetor\n\n");
}


int main() {
    int tam = 10;
    int vetor[10];

    printf("Bem vindo ao sistema de gerenciamento de vetores!\n");
    printf("Para sair do sistema use CTRL + C!!\n\n");

    cadastrarVetor(vetor, tam);
    mostrarVetor(vetor, tam);

    while (1) {
        exibirAcoes();
        int acao = lerInteiro("Digite a ação desejada:\n");

        if (acao == 1) {
            int valorProcurado = lerInteiro("Digite um valor inteiro que deseja procurar no vetor cadastrado!\n");
            int novoValor = lerInteiro("Digite um valor inteiro que deseja substituir o valor procurado no vetor cadastrado!\n");

            int totalAlterado = alterarVetor(vetor, tam, valorProcurado, novoValor);
            if (totalAlterado > 0) {
                printf("Elementos alterados com sucesso, foram alteradas %d posições!\n\n", totalAlterado);
            } else {
                printf("O valor não foi encontrado. Nenhuma posição foi alterada!\n\n");
            }
        } else if (acao == 2) {
            mostrarVetor(vetor, tam);
        } else {
            printf("Valor inserido não é uma operação válida!!\n\n");
        }
    }
}